### PR TITLE
added flag for enable RayTrace finder

### PR DESCRIPTION
**Pull request by saadataz**
Added a flag to enable and disable ray trace focus finder.
This is needed before we tailor the clients to be able to use this feature.

### CODE REVIEW

File: src/focus.service.ts [hunk 1]
```diff
@@ -382,10 +383,8 @@ export class FocusService {
     const directional = isDirectional(direction);
     let element: HTMLElement;
     if (directional) {
-      element = this.findNextFocusByRaycast(direction);
-      if (!element) {
-        element = this.findNextFocus(direction);
-      }
+      if (this.enableRayTrace) { element = this.findNextFocusByRaycast(direction); }
+      if (!element) { element = this.findNextFocus(direction); }
```
Comment (connor4312) [inline src/focus.service.ts]: Line breaks here

**Comment by connor4312** [comment]
QQ: would it make sense to allow ray tracing to be selectively enabled for certain elements? e.g. you could have it disabled by default but enabled for cards on the browse page. This would be a single directive lookup based on the currently selected element.

File: src/focus.service.ts [hunk 2]
```diff
@@ -234,6 +234,7 @@ function isNodeAttached(node: HTMLElement, root: HTMLElement) {
 
 @Injectable()
 export class FocusService {
+  public enableRayTrace = true;
```
Comment (SimonSchick) [inline src/focus.service.ts]: Wouldn't this be something better set via the ctor or a setter? I always feels weird to configure a service by just setting members.

Comment (saadataz) [inline src/focus.service.ts]: this is flexible since we can enable/disable it on different page

Comment (connor4312) [inline src/focus.service.ts]: I would discourage you from following that path. Try to keep state for DOM-related behavior in the DOM.

**Comment by saadataz** [comment]
another directive?
I would rather avoid it for the sake of simplicity.
Nothing beats simplicity :)

**Comment by connor4312** [comment]
In the end it would be simpler than flipping state booleans on essentially a global variable. It's easy to lose track of the state and doesn't truly reflect the intention behind the raycaster -- if you want to raycast, you want to raycast for a particular component or set of components, not for a page. If you use that set of components elsewhere on the site, the raycaster should automatically flip on for those components rather than requiring an "all or nothing" toggle.

**Comment by saadataz** [comment]
Putting things in perspective. This flag is temporary until we adapt our front end to it.

**Comment by connor4312** [comment]
So the flag will be removed in the near future?

**Comment by saadataz** [comment]
yes. like the previous ones

**Comment by connor4312** [comment]
Ok

**Pull request merged**